File: internal/sast/fixtures/javascript/jwt_positive.js
const jwt = require("jsonwebtoken");
const JWT_SECRET = "supersecretpassword12345"; // SC-JS-JWT-003

function decodeUnsafe(token) {
  return jwt.decode(token); // SC-JS-JWT-001
}

function verifyAllowsNone(token) {
  return jwt.verify(token, JWT_SECRET, { algorithms: ['HS256', 'none'] }); // SC-JS-JWT-002
}

File: internal/sast/fixtures/python/jwt_positive.py
import jwt

JWT_SECRET = "supersecretpassword12345"  # SC-PY-JWT-003

def verify_token_unsafe(token):
    return jwt.decode(token, JWT_SECRET, options={"verify_signature": False})  # SC-PY-JWT-001

def verify_token_alg_none(token):
    return jwt.decode(token, JWT_SECRET, algorithms=["none"])  # SC-PY-JWT-002
